examples/step-70/step-70.cc

```diff
@@ -144,6 +144,7 @@ namespace LA
 #endif
 
 #include <cmath>
+#include <filesystem>
```

Comment:
That'll be some fun - For `libstdc++` we will have to figure out a sane way to decide whether we have to link against `libstdc++fs` as well or not.

Reply:
Shouldn't `-std=c++17` imply that?

Reply:
@bangerth gcc version 9 onwards should just require `-std=c++17`, gcc version 8 and older require manual linking with `libstdc++fs`. At least if I recall correctly :smile_cat:

Reply:
I remember that I had similar issues with gcc8. I solved this by substituting them with `boost` libraries:
```
#ifdef __cpp_lib_filesystem
#  include <filesystem>
namespace filesystem
{
  using namespace std::filesystem;
}
#else
#  include <boost/filesystem.hpp>
namespace filesystem
{
  using namespace boost::filesystem;
}
#endif
```
I'm not sure if the `__cpp_lib_filesystem` flag only exists for `gcc`.


We could move this code snippet into the `std_cxx17` namespace. What do you think?

Reply:
The problem is that we don't include the boost filesystem parts in the `bundled/` directory :-( There is also the issue that this is a tutorial and I really didn't want to introduce some complicated workaround in tutorials -- I'd rather just leave things as they are if we can't do it right....